fast64_internal/sm64/sm64_objects.py

```diff
@@ -1403,6 +1421,719 @@ def draw_act(self, obj, layout, value):
+            script_path = decomp_path / "levels" / f"{props.export_level_name}" / "script.c"
+            self.export_level_specific_load(script_path, props)
+        else:
+            script_path = decomp_path / "levels" / "scripts.c"
+            self.export_group_script_load(script_path, props)
+
+    # delims to notify for when to start and end for sig/alt
+    # if match line, then write out to that line
+    # elif fast64_sig, insert after that line
+    # else insert after alt_condition
+    def find_export_lines(
+        self, file_lines, match_str=None, fast64_signature=None, alt_condition=None, start_delim=None, end_delim=None
+    ):
+        search_sig = False if start_delim else True
+        insert_line = 0
+        alt_insert_line = 0
+        match_line = 0
+        for j, line in enumerate(file_lines):
+            if start_delim and start_delim in line:
+                search_sig = True
+                continue
+            if search_sig and match_str and match_str in line:
+                match_line = j
+                break
+            if search_sig and fast64_signature and fast64_signature in line:
+                insert_line = j
+            if search_sig and alt_condition is not None and alt_condition in line:
+                alt_insert_line = j
+            if end_delim and end_delim in line:
+                search_sig = False
+        return match_line, insert_line, alt_insert_line
+
+    # export the model ID to /include/model_ids.h
+    def export_model_id(self, context, props, offset):
+        # check if model_ids.h exists
+        decomp_path = Path(bpy.path.abspath(bpy.context.scene.decompPath))
+        model_ids = decomp_path / "include" / "model_ids.h"
+        if not model_ids.exists():
+            PluginError("could not find model_ids.h")
+
+        model_id_lines = open(model_ids, "r").readlines()
+        export_model_id = f"#define {props.model_id_define: <34}{props.model_id + offset}\n"
+        fast64_sig = "// fast64 object exports get inserted here"
+
+        match_line, sig_insert_line, default_line = self.find_export_lines(
+            model_id_lines,
+            match_str=f"#define {props.model_id_define} ",
+            fast64_signature=fast64_sig,
+            alt_condition="#define MODEL_NONE",
+        )
+
+        if match_line:
+            model_id_lines[match_line] = export_model_id
+        elif sig_insert_line:
+            model_id_lines.insert(sig_insert_line + 1, export_model_id)
+        else:
+            export_line = default_line + 1 if default_line else len(model_id_lines)
+            model_id_lines.insert(export_line, f"\n{fast64_sig}\n")
+            model_id_lines.insert(export_line + 1, export_model_id)
+
+        self.write_file_lines(model_ids, model_id_lines)
+
+    def export_group_script_load(self, script_path, props):
+        if not script_path.exists():
+            PluginError(f"could not find {script_path.stem}")
```

Comment:
A lot of your plugin errors start with lower case, but some of yours and all others in fast64 use normal casing.